### ISSUE TITLE
performance issue when too many folders/files in a diretory

### ISSUE THREAD

**Issue by dropalways** [face-hh/griddycode#8]
my downloads folder got ALOT of shit in it and i was tryna open it and it was lagging heavily

**Comment by dropalways** [comment]
about 21,527 Files, 2,877 Folders

**Comment by lassejlv** [comment]
You got the reason right there if you trying to open a folder with that many files/folders.

**Comment by face-hh** [comment]
To handle such scenarios the `file_manager.gd` script should be updated to load in chunks

**Comment by MinecraftPublisher** [comment]
Just fix this by slicing the file picker's choices down to a limit like 10 before and after the current choice or any other sane number

**Comment by MinecraftPublisher** [comment]
@face-hh do you want me to make a PR for this and you to close this issue aswell? the fix should be easy

**Comment by MinecraftPublisher** [comment]
UPDATE: By limiting item loading to 21 at a time and patching the focus_on to be compatible with this new change, I have significantly reduced the lag of the file picker to the state that it now actually loads your items and works at like 2 fps (further fixes coming soon)

**Comment by EmaBixD** [comment]
This happens even when the file has many lines. (tested on 30,000+)

**Comment by rathlinus** [comment]
Yep, learned that the hard way by opening my 200gb downloads folder
